### PR TITLE
Add new help categories to remove parentheses (context) from help texts

### PR DESCRIPTION
**Pull request by Niloth-p**
### What does this PR do, and why?
Adds the following new help categories to remove the corresponding context-suffixes (in parentheses) from help texts:
- Stream information ("from stream information" suffixes)
- Message information ("from message information" suffixes)
- User list actions ("from user list" suffixes)

That removes parantheses from all the help texts.

Each of the categories are added in their own commit, for readability.
A new command entry `NARROW_TO_USER_PM` is also added to the user list actions.

### Outstanding aspect(s)
This PR is about removing the contexts mentioned in parentheses as suffixes. And it removes all of them.
But, it also introduces new parentheses for the help category names itself.

I went with this because
1. There is not much repetition of the content in the parentheses
2. Length constraints apply to help texts, they don't wrap. But, help category names do wrap.
3. They add some needed clarity, on when/where the help category's hotkeys will work.

Feedback welcome!

### External discussion & connections
- [x] Discussed in **#zulip-terminal** in [`re-categorize`](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Re-Categorization.20of.20Hotkeys/near/1793697)
- [ ] Fully fixes #
- [ ] Partially fixes issue #
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?
- [ ] Manually - Behavioral changes
- [x] Manually - Visual changes
- [ ] Adapting existing automated tests
- [ ] Adding automated tests for new behavior (or missing tests)
- [x] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [x] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [x] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [ ] It contains test additions for any new behavior
- [x] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual changes    
The information popup categories:
image - info popups
The user list actions category:
image - user list actions

### CODE REVIEW

**Review by rsashank — COMMENTED** [review]
LGTM. Seems clear :)

File: zulipterminal/config/keys.py
Comment (zormit) [inline zulipterminal/config/keys.py:323]: This one is a bit tricky because we don't actually use this specific key, but if I understand it correctly the functionality to narrow to a DM happens through the fact that the user list is composed of `UserButton`s, which is a `TopButton`, which maps `ENTER` to `activate` [here](https://github.com/zormit/zulip-terminal/blob/e5e9010850ab4c678890686e3707d7e3e842385e/zulipterminal/ui_tools/buttons.py#L123).

I would suggest to leave a comment here to explain this in fewer words than I do, in order to prevent confusing future readers of this code. Maybe something like "this key is only for documenting the functionality, but the functionality happens through a generic ENTER"?

**Review by zormit — APPROVED** [review]
Generally looks good & clear to me!

I have left one PR comment of something that could be clarified by adding a comment in the code. Apart from that all is approved, so I don't mind if the contributor fixes this or the maintainer adds the comment when merging.

**Comment by Niloth-p** [comment]
Rebased onto main, added a comment as per feedback.

File: zulipterminal/config/keys.py
Comment (zormit) [reply zulipterminal/config/keys.py:337]: @Niloth-p it seems you have now added a comment for `USER_INFO`, which might make sense as well to some extent, but that one is actually used to retrieve its key, e.g. in `zulipterminal/ui_tools/buttons.py`.

Whereas `NARROW_TO_USER_PM` is not used anywhere and just lives here for the documentation/help-text.

Comment (Niloth-p) [reply zulipterminal/config/keys.py:334]: Haha, thanks for the catch. I can't believe I accidentally wrote the comment for the wrong command.

> which might make sense as well to some extent

No, the comment was intended for this command, I wrote it in another text editor and I pasted it into the wrong command. The help text mentions that it clarifies that it does not open the user profile info, and I pasted it for the user info command lol.

Thank you for the save!

**Comment by Niloth-p** [comment]
Rebased and pushed.

We just have 2 keys for the 2 categories 'Stream message info actions' and 'User list actions'. As discussed, those make it tiny categories. So,
- we could either discuss the alternative options that I previously suggested in the CZO topic,
- or leave this PR until more hotkeys are added to these categories,
- or we could push these changes but put these tiny categories at the bottom of the help menu.

**Comment by neiljp** [comment]
@Niloth-p Merging this now; thanks for the earlier review zormit :+1:

My main concern is the location of the resulting extra categories (and their size, as discussed before), but I think this does read more cleanly :)

File: zulipterminal/config/keys.py
Comment (neiljp) [inline zulipterminal/config/keys.py:475]: Ah, I meant to point out here that the f-string should not be necessary for the second string :)

**Pull request merged**
